src/lib/ITensorMPS/src/mpo.jl

```diff
@@ -874,7 +875,6 @@ function ITensors.contract(
     mindim,
     kwargs...,
   )
-  truncate!(C; kwargs...)
```

Comment:
Is there a reason why this line was deleted?

Reply:
Oh yeah, sorry I meant to add a comment here. I'm not really sure what the point of this extra call to truncate is especially seeing as the `maxdim` and `cutoff` kwargs are already spoken for and can't get passed to `truncate`. From looking at the code it appears that `truncate!(psi)` with no extra arguments is equivalent to `orthogonalize!(psi, 1)`, is this the intention?

Reply:
Hmm it's a good question, it may be performing an extra orthogonalization while also making sure the bonds get truncated as much as possible. I think the idea is that zip up isn't an optimal truncation, so `truncate!` can do some extra truncation using the proper orthogonal gauge. @emstoudenmire does that sound right?

Reply:
It looks like that "return sweep" of truncations is mentioned on page 6 of https://arxiv.org/abs/1002.1305.

Reply:
What I get from reading that section is that the initial "zipup" application should be done with a loose tolerance (or large bond dimension) and then since "zipup" leaves the result left orthogonalized it is efficient to do a further truncation sweep down to the target bond dimension.

This makes sense to me, but I don't get how that connects to the ITensor implementation. From what I can tell if you call

```julia
apply(H, psi; maxdim, cutoff)
```

that will perform the "zipup" algorithm with the given parameters and then call `truncate!(C)` which will do no further truncation and "waste" time right orthogonalizing the state. My preference would be to get rid of this call to `truncate!` and document that the "zipup" algorithm is not the most precise and should be done with a large tolerance and then truncated. Alternatively we could add more keyword arguments like `zipup_maxdim` and `zipup_cutoff` to specify the parameters of the "zipup" sweep but that seems a bit clunky to me.

Reply:
It's a good point about maybe having different `cutoff` and `maxdim` options for the zip up sweep and the `truncate!` call. We could have a keyword argument `truncate_kwargs` which is a `NamedTuple` of keyword arguments to pass to `truncate!`, so it would be called like this:
```julia
contract(A, B; alg="zipup", cutoff=1e-9, truncate_kwargs=(; cutoff=1e-5, maxdim=50))
```

However, in your comment you are assuming the call to `truncate!` won't do further truncation even if the same `cutoff` is used by the zip up sweep and the truncation sweep. I don't think that's necessarily true, since the truncation performed by `truncate!` uses the proper orthogonal gauge of the MPS/MPO, unlike the zip up sweep, so it has more information about what states to truncate and can therefore truncate to a smaller bond dimension given a certain cutoff.

Reply:
Oh I see, I didn't notice that the keyword arguments like `cutoff` and `maxdim` weren't being passed to `truncate!`, I think that was a mistake and we meant to pass them.